apps/chat/src/components/Chat/ModelList.tsx

```diff
@@ -185,6 +303,52 @@ export const ModelList = ({
   disabled,
   isReplayAsIs,
 }: ModelListProps) => {
+  const dispatch = useAppDispatch();
+
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState<boolean>(false);
+  const [currentEntityName, setCurrentEntityName] = useState('');
+  const [currentEntityId, setCurrentEntityId] = useState('');
+  const [currentEntityReference, setCurrentEntityReference] = useState('');
+  const [isPublishing, setIsPublishing] = useState<boolean>(false);
+  const recentModelsIds = useAppSelector(ModelsSelectors.selectRecentModelsIds);
+  const applicationDetail = useAppSelector(
+    ApplicationSelectors.selectApplicationDetail,
+  );
+
+  const [entity, setEntity] = useState<ShareEntity | null>(null);
+
+  useEffect(() => {
+    if (currentEntityName && currentEntityId) {
+      setEntity({
+        name: currentEntityName,
+        id: ApiUtils.decodeApiUrl(
+          currentEntityId ||
+            (currentEntityName as unknown as { application: string })
```

Comment:
need to update types here, to be able to read application names without this hack `as unknown as {`